lib/src/commands/create/commands/create_subcommand.dart

```diff
@@ -117,15 +117,43 @@ abstract class CreateSubCommand extends Command<int> {
 
   /// Gets the output [Directory].
   Directory get outputDirectory {
-    final directory = argResults['output-directory'] as String? ?? '.';
-    return Directory(directory);
+    final directory = argResults['output-directory'] as String?;
+
+    if (directory != null) {
+      return Directory(directory);
+    }
+
+    final args = argResults.rest;
+
+    return Directory(args.first);
```

Comment:
same logic as before, if the --output-directory argument isn't provided then treat the [project-name] argument as as a directory path.